### PR TITLE
Add more best practices.

### PR DESCRIPTION
**Pull request by rcorral**

### CODE REVIEW

File: BEST_PRACTICES.md [hunk 1]
```diff
@@ -1,14 +1,89 @@
...
+
+**Event Callbacks**
+
+We split events into two groups. Server and User triggered events. Server events are prefixed with `on` and User events are prefixed with `handle`.
+
+**Examples:**
+* `onMarathonStoreChange` - `MarathonStore` is the subject, while `Change` is the Event.
+* `onVisibilityChange` - Triggered when the browser tab becomes inactive
+* `handleButtonClick` - Handles click on a button
+* `handleImageHover`
```
Comment (philipnrmn) [inline BEST_PRACTICES.md]: - Handles the user hovering an image

**Comment by philipnrmn** [comment]
I like that the methods are grouped logically rather than purely alphabetically. Within the logical groups do they need to be alphabetised?

**Comment by rcorral** [comment]
I don't think we've enforced that yet.

**Comment by philipnrmn** [comment]
It's not enforceable by lint, hence me asking. I occasionally get picked up on it in PRs, but not always.

**Pull request merged**